### PR TITLE
Linting error fixes in test module

### LINKED ISSUE #684: [Dev] fix Python linter errors
The linter test can be started with the following command (also see [development.md](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)):
```bash
./dev/lint.sh
```

We've been ignoring the errors from this script for a while because it has not been added to Travis CI. This ticket is for fixing existing linter errors in the project codebase.

### PR DESCRIPTION
**Pull request by kautukkundan**
(Thanks for sending a pull request! Please make sure to read the contribution guidelines, then fill out the blanks below.)

What changes were proposed in this pull request?
------------------------------------------------
Remaining linting errors in test module 

Does this close any currently open issues?
------------------------------------------
closes #684

### CODE REVIEW

File: e2e_tests/aws_sagemaker/test_sagemaker_update_deployment.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def test_sagemaker_update_deployment(basic_bentoservice_v1, basic_bentoservice_v
 
         request_success, prediction_result = send_test_data_to_endpoint(endpoint_name)
         assert request_success, 'Failed to make successful Sagemaker request'
-        assert '"cat"\n', 'Sagemaker prediction result mismatches with expected value'
+        assert prediction_result == "cat", 'Sagemaker prediction result mismatch'
```
Comment (kautukkundan) [inline e2e_tests/aws_sagemaker/test_sagemaker_update_deployment.py:42]: Exceeds length limit

File: e2e_tests/cli_operations.py [hunk 2]
```diff
@@ -23,6 +23,7 @@ def delete_deployment(deployment_type, deployment_name):
     ) as proc:
         delete_deployment_stdout = proc.stdout.read().decode('utf-8')
     logger.info(delete_deployment_stdout)
+    return delete_deployment_stdout
```
Comment (kautukkundan) [inline e2e_tests/cli_operations.py:27]: This function is called here https://github.com/bentoml/BentoML/blob/master/e2e_tests/yatai_server/test_postgres_s3.py#L47 and a value is assigned to the return, however this didn't return anything

File: e2e_tests/cli_operations.py [hunk 3]
```diff
@@ -36,3 +37,4 @@ def delete_bento(bento_name):
     ) as proc:
         delete_bento_stdout = proc.stdout.read().decode('utf-8')
     logger.info(delete_bento_stdout)
+    return delete_bento_stdout
```
Comment (kautukkundan) [inline e2e_tests/cli_operations.py:40]: This function is called here https://github.com/bentoml/BentoML/blob/master/e2e_tests/yatai_server/test_postgres_s3.py#L47 and a value is assigned to the return, however this didn't return anything

File: e2e_tests/conftest.py [hunk 4]
```diff
@@ -30,11 +30,11 @@ def iris_clf_service():
     clf.fit(X, y)
 
     logger.debug('Creating iris classifier BentoService bundle..')
-    iris_clf_service = IrisClassifier()
-    iris_clf_service.pack('clf', clf)
-    iris_clf_service.save()
+    iris_clf_service_ = IrisClassifier()
+    iris_clf_service_.pack('clf', clf)
+    iris_clf_service_.save()
```
Comment (kautukkundan) [inline e2e_tests/conftest.py:36]: variable name matched function name

File: tests/bento_service_examples/bento_service_with_modified_sys_path.py [hunk 5]
```diff
@@ -7,6 +7,8 @@
 
 sys.path.append(path.dirname(path.abspath(__file__)))
 
+# noqa # pylint: disable=import-error
```
Comment (parano) [inline tests/bento_service_examples/bento_service_with_modified_sys_path.py:10]: 👍

File: e2e_tests/cli_operations.py (hunk 3 above)
Comment (parano) [reply e2e_tests/cli_operations.py:40]: Thanks for surfacing this @kautukkundan 

@yubozhao could you look into it after this PR? It looks like this function was not used properly

Comment (yubozhao) [reply e2e_tests/cli_operations.py:40]: Yep

**Pull request merged**